packages/renderer/src/stores/event-store.ts

```diff
@@ -255,11 +267,20 @@ export class EventStore<T> {
       }
     };
 
-    this.windowEvents.forEach(eventName => {
-      window.events?.receive(eventName, (args?: unknown) => {
-        update(eventName, args as unknown[]).catch((error: unknown) => {
-          console.error(`Failed to update ${this.name}`, error);
-        });
+    this.windowEvents.forEach(eventNameWithOptionalKey => {
+      const [eventName, key] = eventNameWithOptionalKey.split(':');
+      window.events?.receive(eventName, (...args: unknown[]) => {
+        if (key) {
+          if (args.length === 1 && isWithStringKey(args[0]) && key === args[0].key) {
+            update(eventNameWithOptionalKey, args).catch((error: unknown) => {
+              console.error(`Failed to update ${this.name}`, error);
+            });
+          }
+        } else {
+          update(eventNameWithOptionalKey, args).catch((error: unknown) => {
+            console.error(`Failed to update ${this.name}`, error);
+          });
+        }
```

Comment:
Here, we are potentially registering several listeners for the same eventName (if we pass `eventName:key1`, `eventName:key2`, `eventName:key3` to `windowEvents`).

If this is problematic, I can regroup all window events with a common eventName, and start a single listener

Reply:
could use eventName:key1,key2,key3 ? (assuming a key can't have any comma)